.github/workflows/ci.yml

```diff
@@ -0,0 +1,63 @@
+name: CI
+
+on:
+  pull_request:
+  push:
+    branches:
+      - master
+    tags:
+
+jobs:
+  test:
+    name: ${{ matrix.os }} / ${{ matrix.python-version }} / ${{ matrix.pip-version }}
+    runs-on: ${{ matrix.os }}-latest
+    strategy:
+      matrix:
+        os:
+          - Ubuntu
+          - Windows
+        python-version:
+          - 3.8
+          - 2.7
+          - pypy3
+          - 3.5
+          - 3.6
+          - 3.7
+          - pypy2
+        pip-version:
+          - "latest"
+          - "20.0"
+        # Test pypy only with the latest pip version to reduce the number of jobs
+        exclude:
+          - pip-version: "20.0"
+            python-version: pypy2
+          - pip-version: "20.0"
+            python-version: pypy3
+          # TODO: fix test_realistic_complex_sub_dependencies test
+          #       and remove Windows exclusion
+          - os: Windows
+            python-version: pypy2
+          - os: Windows
+            python-version: pypy3
+
+    env:
+      PY_COLORS: 1
+      TOXENV: pip${{ matrix.pip-version }}-coverage
+      TOX_PARALLEL_NO_SPINNER: 1
+    steps:
+      - uses: actions/checkout@master
+      - name: Set up Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v2
+        with:
+          python-version: ${{ matrix.python-version }}
+      - name: Install tox
+        run: pip install tox
+      - name: Prepare test environment
+        run: tox --notest -p auto --parallel-live
+      - name: Test pip ${{ matrix.pip-version }}
+        run: tox
+      - name: Upload coverage to Codecov
+        uses: codecov/codecov-action@v1.0.6
+        with:
+          file: ./coverage.xml
+          name: ${{ runner.os }}-${{ matrix.python-version }}-${{ matrix.pip-version }}
```

Comment:
Does python package work without a token? The action can do this. Also, it's better to have an officially maintained piece of the puzzle that is designed for this ecosystem instead of reinventing the wheel.

Reply:
@webknjaz hm… both versions are maintained by Codecov, so I don't see a big problem in that regard. However, you do need a token. I am just saying its fast, as in a lot faster. If you don't care waiting for CI this should be fine.

Reply:
Weird. It shouldn't be slower because it's a JS action, not docker.

Reply:
Oh, have you tried it after Nov 19? This is when they switched from Docker to JS.

Reply:
> both versions are maintained by Codecov

But maintained here I meant "and designed for use within the Actions ecosystem"

Reply:
<img width="655" alt="Screenshot 2020-05-04 at 16 15 31" src="https://user-images.githubusercontent.com/7377671/80951722-98b12580-8e22-11ea-8932-1ac4f4b036a7.png">

Four seconds to get job done - it's pretty good in my opinion.